### DISCUSSION TITLE
Use Supabase Queries Inside Supabase Queries, but Run on Server

### DISCUSSION THREAD

**Discussion by jdgamble555** [supabase/supabase#13329] [Feature Requests]
Chat GPT told me I could do this and it resolves on the server:

```ts
  const { data, error } = await supabase
    .from('Posts')
    .select('id, content, createdAt')
    .filter('userId', 'in', supabase
      .from('Followers')
      .select('following')
      .eq('follower', $UID)
    )
    .order('createdAt', { ascending: false });
```

Obviously it doesn't work, but it would be a great idea!

J

**Comment by rexmondo** [comment]
You can use the [embedded resource language from postgrest](https://postgrest.org/en/stable/api.html#resource-embedding)--the server that supabase runs--to do this. I do it all the time and it works quite well.

It looks like you're trying to get posts for the users the user is following. You could do something along the line of:

```
const result = await supabase
  .from('Followers')
  .select('follower, following(posts(id, content, createdAt))')
  .eq('follower', $UID)
```

assuming that:

- following is a relationship column
- the Posts table has a user relationship like author or something
- all the foreign key columns meet the correct constraints (ie on many-to-many, the foreign keys need to be part of a composite primary key)
- youre willing to unpack the posts in the nested data and pass them back so it looks like you queried via the posts table

Embedded resources are basically supabase/postgrest's way of doing joins. If you get good at them, they give you many of the conveniences of something like graphql.

**Comment by jdgamble555** [comment]
No, I get that. Thank you. Just thought it was interesting to be able to overlap the JS calls like that. Chat GPT thought that was how it was done.

J
